artifact/v3/manifest/manifest.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_ARTIFACT_V3_MANIFEST_PARSER_HPP
+#define MENDER_ARTIFACT_V3_MANIFEST_PARSER_HPP
+
+#include <string>
+#include <unordered_map>
+
+#include <common/expected.hpp>
+#include <common/error.hpp>
+#include <common/io.hpp>
+
+namespace mender {
+namespace artifact {
+namespace v3 {
+namespace manifest {
+
+using namespace std;
+
+namespace expected = mender::common::expected;
+namespace io = mender::common::io;
+namespace error = mender::common::error;
+
+struct Manifest {
```

Comment:
I don't like `struct`s with functions (methods), but YMMV.

Reply:
I'll have a look in the style-guide. I'm fine either way

Reply:
I guess it's a little bit up for impretation:

`Use a struct only for passive objects that carry data; everything else is a class.`

https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes

It is made just for carrying data, but it has an `interface` through `Get`. I'm guessing I'll favour your judgement and go with class :+1: